node.go

```diff
@@ -0,0 +1,176 @@
+package merkletree
+
+import (
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+type interiorNode struct {
+	parent     *interiorNode
+	leftChild  MerkleNode
+	rightChild MerkleNode
+	leftHash   []byte
+	rightHash  []byte
+	level      int
+}
```

Comment:
Making `interiorNode` the base is kind of confusing in that leafs and empty nodes don't have children, right?

I'd suggest making a type

    type Node struct {
      level int
      parent *node
    }

that they all share.  Or, even making `emptyNode` the base, but then taking care to always use that as the default in your switches.